### ISSUE TITLE
SDK component that is missing or mismatched.

### ISSUE THREAD

**Issue by Ali1Jamal** [discosultan/dx12-game-programming#6]
SharpDX.SharpDXException: 'HRESULT: [0x887A002D], Module: [SharpDX.DXGI], ApiCode: [DXGI_ERROR_SDK_COMPONENT_MISSING/SdkComponentMissing], Message: The application requested an operation that depends on an SDK component that is missing or mismatched.

**Comment by discosultan** [comment]
Thanks for reporting! I'm also experiencing this. This is probably due to some change in recent Windows update.

As a quick workaround, you can try running the samples in Release mode.

**Edit:** I found out some more information on the topic from [this thread](https://github.com/Microsoft/DirectX-Graphics-Samples/issues/54).

The problem is that the Direct3D 12 debug layer is not present by default. This can be checked by ensuring a `C:\Windows\System32\d3d12SDKLayers.dll` file is present.

The debug layer can be added through settings page "Manage optional features" > "Add a feature" > "Graphics Tools".

I'll add a check to this project only to enable debug layer if it is present, so it wouldn't crash like this anymore.

**Comment by discosultan** [comment]
Should be fixed by https://github.com/discosultan/dx12-game-programming/commit/d3b2042c192657ba87d1828f79c8fa603ec67369.

Please leave a comment or open a new issue if the problem persists.

**Comment by Ali1Jamal** [comment]
now give me this error:
SharpDX.SharpDXException: 'HRESULT: [0x80004005], Module: [General], ApiCode: [E_FAIL/Unspecified error], Message: Unspecified error
'

**Comment by discosultan** [comment]
Can you provide me some more context:

- Which sample are you trying to execute?
- Which line throws the error?

**Comment by Ali1Jamal** [comment]
ok
iam trying to execute Skinned mesh.
in D3DApp.cs line 428 ......                Device = new Device(null, FeatureLevel.Level_11_0);

**Comment by discosultan** [comment]
Thanks for the info. The skinned mesh sample is working fine for me.

I'm curious about the error because it is raised in the shared library used by all the samples. Is the error present in every sample or just the skinned mesh one? If it's present in all of them, make sure your GPU supports Direct3D feature level 11.

Alternatively, you could try running the same sample from the original C++ samples repo found [here](https://github.com/d3dcoder/d3d12book). If it works there, then we know the error is somewhere in my code.

**Comment by Ali1Jamal** [comment]
ok i have the error in all samples.and my GPU its intel hd 4000 and its support dx11 and dx 12 .i wall try to download visual c++ then try the samples.
thank you for your time

**Comment by discosultan** [comment]
Right, as seen [here](https://en.wikipedia.org/wiki/Feature_levels_in_Direct3D), ~Intel HD 4000 should support feature level 11_0~~ under the Direct3D 12 section, Intel HD 4200 is the earliest hardware to support feature level 11_0 which the samples are using.

~~Let me know if the C++ samples work for you. I'm curious to find out what the issue is.~~

**Comment by Ali1Jamal** [comment]
Thank you for your quick responses.
see this image
https://imgur.com/aSr4JUH
and i wall try c++

**Comment by discosultan** [comment]
As far as I know, the DirectX version in `dxdiag` shows the version of installed graphics software but not the actual Direct3D level supported by hardware.

It seems there's not a very convenient way to check for that support. I found a relevant thread [here](https://superuser.com/questions/44952/how-can-i-determine-the-highest-gpu-supported-directx-version).

**Comment by Ali1Jamal** [comment]
Right.
https://imgur.com/MyPXSl2
i am waiting c++ installing because its take long

**Comment by Ali1Jamal** [comment]
ok i am try the c++ samples and its work fine.
https://imgur.com/k5Bak7F

**Comment by discosultan** [comment]
Thanks for trying out the C++ samples, @Ali1Jamal. The samples should fallback to a CPU emulation if proper graphics hardware is not present. That's also the reason you have 1 FPS in your screenshot. However, this does not seem to work in C# samples. I opened a new issue: https://github.com/discosultan/dx12-game-programming/issues/7

**Edit:** The samples should now work with CPU emulation. Fixed in https://github.com/discosultan/dx12-game-programming/commit/ed3429fbc00331af09afbbf9fb4767fc8db5b69c
